### PR TITLE
[79, 67, 72] Home landing page

### LINKED ISSUE #67: Switching Role might go unnoticed
### Problem

This might be amplified by BUG #66, but whenever I switch role, I would like to have a feeling for it.

### Feature Description

Improve the UX when switching role:
- we could change background color theme to another
- we could add an icon top left on top of the side menu
- ...

### LINKED ISSUE #72: Home: dashboard
### Problem

We do not have a Home screen at the moment.

### Feature Description

Design a dashboard with some widgets to the app, or external content.

Eventually remove it.

### Screenshots

image

### LINKED ISSUE #79: SidebarContent: counters
### Describe the bug

There are counters associated to some menu entries. The endpont APIs seem to work replying with a total number of items.

https://github.com/tupuio/tupuapp/blob/a6ac1aa6ec1a561cb199faa78d155c59c6aa65f3/components/Sidebar/SidebarContent.js#L19-L26

The incoming data though is left `undefined`, and the counters are always set to `0`.

We query the endpoints at each page load (or are they cached by `useSWR`?), also the ones not necessary for the menu (given by `mode`).

A note: those counters API endpoints should only return the new items, not all of them. I think it's a bigger problem to be solved in this report.

### To Reproduce

1. Go to home page

### Expected behavior

A menu item is showing the total number of entries for that menu (e.g. `Applications`, total number of applications).

### Actual behavior

The counters (see `tag`) are never shown, they are always `0`.

### PR DESCRIPTION
**Pull request by lorepirri**
### What Github issue does this PR relate to?

Changes proposed in this pull request:

- Close #79 
- Close #67 
- Close #72 

image

---

image

### CODE REVIEW

File: components/Sidebar/SidebarContent.js [hunk 1]
```diff
@@ -70,9 +68,19 @@ const SidebarContent = ({ onClose, mode, ...rest }) => {
...
+        </Text>
+      </Flex>
       <Box mt={50}>
         {links.map((link) => (
-          <NavItem key={link.name} icon={link.icon} href={link.href}>
+          <NavItem 
+                  key={link.name} 
+                  icon={link.icon} 
+                  href={link.href}
+                  bg={mode === "mentor"?"brand.blue":"brand.green"}
```
Comment (raluka) [inline components/Sidebar/SidebarContent.js:82]: Looks like we reuse this logic for the background setup. Maybe it makes sense to extract it in a reusable variable

**Review by raluka — APPROVED** [review]
Looks good 😸

**Review by maranta1 — COMMENTED** [review]
Because we're requesting the `requestsData`, `menteesData`, `mentorshipsData` and `applicationsData` objects asynchronously in pages/index.js, these objects are not defined when we execute the logic to decide the path. This means that the default path is always loaded. We could either simplify the logic to always load the default path, or refactor to wait for all of the requests? Something like the below could work:

```js
import useSWR from "swr";
import { useRouter } from 'next/router'

const fetcher = url => fetch(url).then(response => response.json())

const multipleURLfetcher = (urls) => {

  const urlArray = Object.keys(urls);

  return Promise.all(urlArray.map(urlKey => fetcher(urlKey))).then((responses) => responses.reduce((returnObj, currentResponse, index) => {
    // get the response label using the array of URLs passed to Promise.all function
    // and the original urls object
    const responseLabel = urls[urlArray[index]]

    // assign the response to the r
    returnObj[responseLabel] = currentResponse
    return returnObj
  }, {}))
}

export default function Home({ mode }) {
  const router = useRouter();

  const { data, isValidating } = useSWR({
    "/api/requestsCount": "requestsData",
    "/api/menteesCount": "menteesData",
    "/api/profile": "profile",
    "/api/applicationsCount": "applicationsData",
    "/api/mentorshipsCount": "mentorshipsData"
  }, multipleURLfetcher)

  if (!isValidating) {
    const { requestsData, menteesData, profile, applicationsData, mentorshipsData } = data


    let homePath = '/profile';
    if (mode === "mentor") {
      const requestsCount = requestsData?.count || 0;
      const menteesCount = menteesData?.count || 0;

      if (profile && !profile?.mentor?.calendly) {
        // will go to the profile to let the mentor set their calendar link
        homePath = '/preferences';
      } else if (requestsCount) {
        homePath = '/requests';
      } else if (menteesCount) {
        homePath = '/mentees';
      }
    } if (mode === "mentee") {
      const applicationsCount = applicationsData?.count || 0;
      const mentorshipsCount = mentorshipsData?.count || 0;

      if (mentorshipsCount) {
        homePath = '/mentorships';
      } else if (applicationsCount) {
        homePath = '/applications';
      } else {
        homePath = '/mentors';
      }
    }

    router.push(homePath);
    return <></>;
  }

  return <>Loading...</>
}
```
WDYT?

File: components/Sidebar/SidebarContent.js (hunk 1 above)
Comment (lorepirri) [reply components/Sidebar/SidebarContent.js:82]: totally right, it will be part of a clean up later on

**Comment by lorepirri** [comment]
> Because we're requesting the `requestsData`, `menteesData`, `mentorshipsData` and `applicationsData` objects asynchronously in pages/index.js, these objects are not defined when we execute the logic to decide the path. This means that the default path is always loaded. We could either simplify the logic to always load the default path, or refactor to wait for all of the requests? Something like the below could work:
> 
> ```js
> import useSWR from "swr";
> import { useRouter } from 'next/router'
> 
> const fetcher = url => fetch(url).then(response => response.json())
> 
> const multipleURLfetcher = (urls) => {
> 
>   const urlArray = Object.keys(urls);
> 
>   return Promise.all(urlArray.map(urlKey => fetcher(urlKey))).then((responses) => responses.reduce((returnObj, currentResponse, index) => {
>     // get the response label using the array of URLs passed to Promise.all function
>     // and the original urls object
>     const responseLabel = urls[urlArray[index]]
> 
>     // assign the response to the r
>     returnObj[responseLabel] = currentResponse
>     return returnObj
>   }, {}))
> }
> 
> export default function Home({ mode }) {
>   const router = useRouter();
> 
>   const { data, isValidating } = useSWR({
>     "/api/requestsCount": "requestsData",
>     "/api/menteesCount": "menteesData",
>     "/api/profile": "profile",
>     "/api/applicationsCount": "applicationsData",
>     "/api/mentorshipsCount": "mentorshipsData"
>   }, multipleURLfetcher)
> 
>   if (!isValidating) {
>     const { requestsData, menteesData, profile, applicationsData, mentorshipsData } = data
> 
> 
>     let homePath = '/profile';
>     if (mode === "mentor") {
>       const requestsCount = requestsData?.count || 0;
>       const menteesCount = menteesData?.count || 0;
> 
>       if (profile && !profile?.mentor?.calendly) {
>         // will go to the profile to let the mentor set their calendar link
>         homePath = '/preferences';
>       } else if (requestsCount) {
>         homePath = '/requests';
>       } else if (menteesCount) {
>         homePath = '/mentees';
>       }
>     } if (mode === "mentee") {
>       const applicationsCount = applicationsData?.count || 0;
>       const mentorshipsCount = mentorshipsData?.count || 0;
> 
>       if (mentorshipsCount) {
>         homePath = '/mentorships';
>       } else if (applicationsCount) {
>         homePath = '/applications';
>       } else {
>         homePath = '/mentors';
>       }
>     }
> 
>     router.push(homePath);
>     return <></>;
>   }
> 
>   return <>Loading...</>
> }
> ```
> 
> WDYT?

agree, on the preview seems to work, even though it should not.

I created a followup ticket here #119

I will merge this and let's discuss later about it and see what's best!

**Comment by lorepirri** [comment]
thanks @raluka and @maranta1 for reviewing this! I will merge now.

**Pull request merged**